pontoon/base/views.py

```diff
@@ -314,13 +314,28 @@ def get_translations_info(translations):
         translations.update(
             approved=True,
             approved_user=request.user,
-            approved_date=timezone.now()
+            approved_date=timezone.now(),
+            rejected=False,
+            rejected_user=None,
+            rejected_date=None,
         )
 
-    elif action == 'delete':
-        count, translated_resources, changed_entities = get_translations_info(translations)
-        TranslationMemoryEntry.objects.filter(translation__in=translations).delete()
-        translations.delete()
+    elif action == 'reject':
+        suggestions = Translation.objects.filter(
+            locale=locale,
+            entity__pk__in=entities,
+            approved=False,
+            rejected=False
+        )
+        count, translated_resources, changed_entities = get_translations_info(
+            suggestions
+        )
+        TranslationMemoryEntry.objects.filter(translation__in=suggestions).delete()
+        suggestions.update(
+            rejected=True,
+            rejected_user=request.user,
+            rejected_date=timezone.now(),
+        )
```

Comment:
Mind if I do that in a follow-up bug? This is getting pretty big, and I plans to refactor this batch actions logic. I could do that there.